File: tool/tool/ViewController.m
//
//  ViewController.m
//  tool
//
//  Created by 王博 on 2017/10/20.
//  Copyright © 2017年 NewsInPalm. All rights reserved.
//

#import "ViewController.h"
#import <AdSupport/AdSupport.h>
@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITextField *idfatext;
@property (weak, nonatomic) IBOutlet UITextField *Otext;
@property (weak, nonatomic) IBOutlet UITextField *subidtext;
@property (weak, nonatomic) IBOutlet UITextField *urltext;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.idfatext.text = [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
}
- (IBAction)didclick:(id)sender {
    self.urltext.text = [NSString stringWithFormat:@"http://adnetwork.hadoopfinetech.com/adserver/click?a=100003&o=%@&advid=%@&subid=%@&affsub=",self.Otext.text,self.idfatext.text,self.subidtext.text];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: ScrollView布局/ScrollView布局/ViewController.swift
//
//  ViewController.swift
//  ScrollView布局
//
//  Created by 王博 on 17/4/28.
//  Copyright © 2017年 NewsInPalm. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var contentView: UIView!
    @IBOutlet weak var redView: UIView!
    @IBOutlet weak var widthConstraint: NSLayoutConstraint!
    @IBOutlet weak var heightConstraint: NSLayoutConstraint!

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        widthConstraint.constant = 2 * view.bounds.size.width
        heightConstraint.constant = 2 * view.bounds.size.height
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

